app/src/main/java/com/ik/exploringviewmodel/base/BaseLifecycleActivity.kt

```diff
@@ -15,14 +14,9 @@ abstract class BaseLifecycleActivity<T : AndroidViewModel> : AppCompatActivity()
 
     abstract val viewModelClass: Class<T>
 
-    protected lateinit var viewModel: T
+    protected val viewModel by lazy { ViewModelProviders.of(this).get(viewModelClass)!! }
```

Comment:
Thank you 🙏 
Could you use something more safety than ```!!``` ? Could you make viewModel nullable for instance?
Could you use LazyThreadSafetyMode.NONE for lazy??? You can also make this improvement in ReposActivity.